2_process/src/make_lake_metadata_augmented.py

```diff
@@ -0,0 +1,52 @@
+import os
+import requests
+import urllib
+import pandas as pd
+
+
+def elevation_query(lat, lon):
+    """
+    Query service using lat, lon and return the corresponding elevation values.
+    Thanks to https://gis.stackexchange.com/questions/338392/getting-elevation-for-multiple-lat-long-coordinates-in-python
+
+    :param lat: latitude values in decimal degrees
+    :param lon: longitude values
+    :returns: elevation at the specified coordinates
+
+    """
+    # USGS Elevation Point Query Service
+    url = r'https://nationalmap.gov/epqs/pqs.php?'
+    # define rest query params
+    params = {
+        'output': 'json',
+        'x': lon,
+        'y': lat,
+        'units': 'Meters'
+    }
+
+    # format query string and return query value
+    result = requests.get((url + urllib.parse.urlencode(params)))
+    elevation = result.json()['USGS_Elevation_Point_Query_Service']['Elevation_Query']['Elevation']
```

Comment:
This is so cool!

Reply:
Yes, it's a cool service! I will note, though, that querying 881 elevations for 881 lakes is pretty slow. Jordan's working on adding elevation to the metadata in lake-temperature-model-prep so this query won't be needed for the full set of lakes.